### PR TITLE
deploy integration tests based off kubernetes-core + an overlay

### PR DESCRIPTION
**Pull request by addyess**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,7 +13,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python: [3.6, 3.7, 3.8, 3.9]
+        python: 
+          - "3.6"
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
```
Comment (mateoflorido) [inline .github/workflows/main.yml:21]: Could we use the reusable workflows here?

**Review by mateoflorido — COMMENTED** [review]
LGTM!

File: .github/workflows/main.yml (hunk 1 above)
Comment (stonepreston) [reply .github/workflows/main.yml:21]: good thinking, I added them to the worker charm, but didnt get around to this one:
https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/122/files

Comment (addyess) [reply .github/workflows/main.yml]: as you wish

**Review by kwmonroe — APPROVED** [review]

**Pull request merged**